### ISSUE TITLE
BUYShippingRate.shippingRateIdentifier is nil on iphone 5

### ISSUE THREAD

**Issue by mihai-moldovan** [Shopify/mobile-buy-sdk-ios#313]
I'm using shippingRateIdentifier in the UI code to keep track of the selected shipping rate until the user applies the changes. However shippingRateIdentifier is nil on iphone 5 (device and simulator) on all tested ios versions (8.4, 9.3) 
Also it appears that BUYShippingRate.identifier is a String instead of NSNumber:

```
(lldb) p rate.identifier
(NSNumber!) $R4 = (null) {
  Foundation.NSValue = shopify-US:%202-4%20business%20days%20-%20Under%20$4,000-15.00
}
```
I attached a sample project where you can reproduce the issue:  [BuyTest.zip](https://github.com/Shopify/mobile-buy-sdk-ios/files/446546/BuyTest.zip)

Please add your own domain, api key and product id

**Comment by mihai-moldovan** [comment]
This also happens on iphone 4S. I'm using sdk version 2.0

**Comment by bgulanowski** [comment]
This seems to imply that it works on iPhone 5S and better. We don't have any code which checks the OS version. I'm not aware of any code which relies on 64-bit values, although it's possible.

Keep in mind that Shopify objects have identifiers that are 64 bits. We have tried to make everything an NSNumber so that any 64-bit values coming down the wire are handled correctly, but if you have any insight into where this might not be the case, please let us know.

**Comment by mihai-moldovan** [comment]
OK, I investigated more. So I don't think it's related to any platform or OS version, although it easier to reproduce on iphone 5 with os version 8.x. 

Anyway I think I found the problem: ShippingRate entity defines JSONPropertyKey for shippingRateIdentifier as being "id"
[In NSEntityDescription+BUYAdditions.m](https://github.com/Shopify/mobile-buy-sdk-ios/blob/master/Mobile%20Buy%20SDK/Mobile%20Buy%20SDK/Additions/NSEntityDescription%2BBUYAdditions.m#L111) 
```
// json key -> property name
- (NSDictionary *)buy_JSONKeyMapping
{
	return [[self buy_PropertyNameMapping] buy_reverseDictionary];
}
```
The first call will return a dictionary like this:
```
{
    "identifier": "id",
    "shippingRateIdentifier": "id"
}
```
which is then reversed using buy_reverseDictionary. This, in turn will yield an undefined result. 
Sometimes it is:
`{ "id": "shippingRateIdentifier" }`
other times it is:
`{ "id": "identifier" }`

That's why, in the second case, shippingRateIdentifier is nil and identifier is a String  instead of NSNumber.

Do you have any workaround for this?

**Comment by bgulanowski** [comment]
`ShippingRate` should not have an `identifier` property at all. Thank you for finding this.

The solution is to remove the `identifier` property from the `ShippingRate` entity in the managed object model, "Mobile Buy SDK.xcdatamodeld".

I will put up a PR right now, and we should have something merged later today, but we hadn't made explicit plans for when we will cut the next version.

**Comment by bgulanowski** [comment]
Fixed by #319
